### DISCUSSION TITLE
Is the FastAPI documentation giving wrong advice about dependency injection of DB sessions?

### DISCUSSION THREAD

**Discussion by DonDebonair** [fastapi/fastapi#9361] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def get_db():
    db = SessionLocal()
    try:
        yield db
    finally:
        db.close()


@app.post("/users/", response_model=schemas.User)
def create_user(user: schemas.UserCreate, db: Session = Depends(get_db)):
    db_user = crud.get_user_by_email(db, email=user.email)
    if db_user:
        raise HTTPException(status_code=400, detail="Email already registered")
    return crud.create_user(db=db, user=user)
```

### Description

In the FastAPI documentation, [an example is given](https://fastapi.tiangolo.com/tutorial/sql-databases/#create-a-dependency) on how to create a _dependency_ for you database session. That part of the documentation contains a bug: dependencies created with `Depends` are **cached** by default, but the database session is closed in the `finally` block, which is fired at the end of the request. So now we're caching a closed DB session.

Using this example anecdotally has led to bugs in our production system, which seemed to go away when we used `use_cache=False` on `Depends`.

I think this is a footgun that should be very clearly explained in the documentation

### Operating System

Linux

### FastAPI Version

0.89.1

### Python Version

3.10

**Comment by YuriiMotov** [comment, marked as answer]
There is no problem here. At least in the current version.

Dependencies are cached by FastAPI for each request separately. And the `finally` block is executed after endpoint function execution is finished.
